repository/src/main/java/org/jbpm/process/workitem/repository/service/RepoService.java

```diff
@@ -121,17 +124,59 @@ protected void loadModules() {
         }
     }
     
+    /*
+     * Service related operations with resultMap
+     */
+
+    public void addService(RepoData service, Map<String, List<String>> resultMap) {
+        if (!addNewService(service, resultMap)) {
+            updateService(service, resultMap);
```

Comment:
why separating this into to method calls?

Reply:
done